server-spi-private/src/main/java/org/keycloak/services/clientpolicy/ClientPoliciesUtil.java

```diff
@@ -0,0 +1,747 @@
+     * it can be constructed by merging proposed client policies with existing client policies.
+     * not return null.
+     */
+    private static ClientPoliciesRepresentation getValidatedClientPoliciesRepresentation(KeycloakSession session, RealmModel realm, ClientPoliciesRepresentation proposedPoliciesRep) throws ClientPolicyException {
+        if (proposedPoliciesRep == null) {
+            proposedPoliciesRep = new ClientPoliciesRepresentation();
+        }
+        if (realm == null) {
+            throw new ClientPolicyException("realm not specified.", "NA");
+        }
+
+        // deserialize existing profiles (json -> represetation)
+        ClientPoliciesRepresentation existingPoliciesRep = null;
+        String existingPoliciesJson = session.clientPolicy().getClientPoliciesJsonString(realm);
+        if (existingPoliciesJson != null) {
+            try {
+                existingPoliciesRep = convertClientPoliciesJsonToRepresentation(existingPoliciesJson);
+            } catch (ClientPolicyException cpe) {
+                throw new ClientPolicyException(cpe.getError(), cpe.getErrorDetail());
+            }
+            if (existingPoliciesRep == null) {
+                existingPoliciesRep = new ClientPoliciesRepresentation();
+            }
+        } else {
+            existingPoliciesRep = new ClientPoliciesRepresentation();
+        }
+
+        // no policy contained (it is valid)
+        // back to initial builtin policies
+        List<ClientPolicyRepresentation> proposedPolicyRepList = proposedPoliciesRep.getPolicies();
+        if (proposedPolicyRepList == null || proposedPolicyRepList.isEmpty()) {
+            proposedPolicyRepList = new ArrayList<>();
+            proposedPoliciesRep.setPolicies(new ArrayList<>());
+         }
+
+        // duplicated policy name is not allowed.
+        if (proposedPolicyRepList.size() != proposedPolicyRepList.stream().map(i->i.getName()).distinct().count()) {
+            throw new ClientPolicyException("proposed client policy name duplicated.", "NA");
+        }
+
+        // construct updating policies from existing policies and proposed policies
+        ClientPoliciesRepresentation updatingPoliciesRep = new ClientPoliciesRepresentation();
+        updatingPoliciesRep.setPolicies(new ArrayList<>());
+        List<ClientPolicyRepresentation> updatingPoliciesList = updatingPoliciesRep.getPolicies();
+
+        // add existing builtin policies to updating policies
+        List<ClientPolicyRepresentation> existingPoliciesList = existingPoliciesRep.getPolicies();
+        if (existingPoliciesList != null && !existingPoliciesList.isEmpty()) {
+            existingPoliciesList.stream().filter(i->i.isBuiltin()).forEach(i->updatingPoliciesList.add(i));
+        }
+
+        for (ClientPolicyRepresentation proposedPolicyRep : proposedPoliciesRep.getPolicies()) {
+            if (proposedPolicyRep.getName() == null) {
+                continue;
```

Comment:
To me, it will be better to throw the error instead of quietly ignore the invalid policy with the missing name.

Reply:
@mposolda I also agree with you. I would like to fix codes to throw the error as you previously proposed if the proposed policy do not have its name.